src/prefect/workers/process.py

```diff
@@ -186,3 +193,55 @@ async def run(
         return ProcessWorkerResult(
             status_code=process.returncode, identifier=str(process.pid)
         )
+
+    async def kill_infrastructure(
+        self, infrastructure_pid: str, grace_seconds: int = 30
+    ):
+        hostname, pid = _parse_infrastructure_pid(infrastructure_pid)
```

Comment:
Wasn't one of the ideas for work pools / workers that we could drop the first half of the infrastructure PIDs since we expect workers to be tied to a single type of infrastructure?

Reply:
Work pools are tied to a single type of infrastructure, but not necessary a single instance of infrastructure. If there are multiple process workers polling a single work pool, the subprocess executing a flow run could be on any of the process workers, so I think it's worthwhile to keep the `hostname` part of the infrastructure PID.

Reply:
Hm I agree but I thought workers was how we were going to reduce permissions for the Kubernetes worker so it doesn't need to get a cluster identifier? I'm fine with leaving it as-is if that's not the case.